crates/sqlite_store/src/store.rs

```diff
@@ -0,0 +1,893 @@
+
+        let tx_graph: indexed_tx_graph::ChangeSet<A, keychain::ChangeSet<K>> =
+            indexed_tx_graph::ChangeSet { graph, indexer };
+
+        if network.is_none() && chain.is_empty() && tx_graph.is_empty() {
+            Ok(None)
+        } else {
+            Ok(Some(ChangeSet {
+                network,
+                chain,
+                tx_graph,
+            }))
+        }
+    }
+}
+
+impl<K> ReadWrite<K, ConfirmationTimeHeightAnchor> for Store<K, ConfirmationTimeHeightAnchor>
+where
+    K: Ord + for<'de> Deserialize<'de> + Serialize + Send,
+{
+    fn db_transaction(&mut self) -> Result<rusqlite::Transaction, Error> {
+        self.db_transaction()
+    }
+}
+
+impl<K> ReadWrite<K, ConfirmationHeightAnchor> for Store<K, ConfirmationHeightAnchor>
+where
+    K: Ord + for<'de> Deserialize<'de> + Serialize + Send,
+{
+    fn db_transaction(&mut self) -> Result<rusqlite::Transaction, Error> {
+        self.db_transaction()
+    }
+}
+
+impl<K> ReadWrite<K, BlockId> for Store<K, BlockId>
+where
+    K: Ord + for<'de> Deserialize<'de> + Serialize + Send,
+{
+    fn db_transaction(&mut self) -> Result<rusqlite::Transaction, Error> {
+        self.db_transaction()
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::ChangeSet;
+    use bdk_chain::bitcoin::consensus::encode::deserialize;
+    use bdk_chain::bitcoin::constants::genesis_block;
+    use bdk_chain::bitcoin::hashes::hex::FromHex;
+    use bdk_chain::bitcoin::transaction::Transaction;
+    use bdk_chain::bitcoin::Network::Testnet;
+    use bdk_chain::bitcoin::{secp256k1, BlockHash, OutPoint};
+    use bdk_chain::miniscript::Descriptor;
+    use bdk_chain::{
+        indexed_tx_graph, keychain, tx_graph, ConfirmationTimeHeightAnchor, DescriptorExt,
+    };
+    use bdk_persist::PersistBackend;
+    use std::str::FromStr;
+    use std::sync::Arc;
+
+    #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Hash, Debug, Serialize, Deserialize)]
+    enum Keychain {
+        External { account: u32, name: String },
+        Internal { account: u32, name: String },
+    }
+
+    #[test]
+    fn insert_and_load_aggregate_changesets_with_confirmation_time_height_anchor() {
+        let changeset = create_test_changeset(&|block_hash_1| ConfirmationTimeHeightAnchor {
+            confirmation_height: 1,
+            confirmation_time: 1296667328,
+            anchor_block: (1, block_hash_1).into(),
+        });
+
+        let mut store = Store::<Keychain, ConfirmationTimeHeightAnchor>::new_memory()
+            .expect("create new memory db store");
+        // let mut store =
+        //     Store::<Keychain, ConfirmationTimeHeightAnchor>::new(Path::new("test_agg.sqlite"))
+        //         .expect("create new file db store");
+
+        store.write_changes(&changeset).expect("write changeset");
+
+        let agg_changeset = store.load_from_persistence().expect("aggregated changeset");
+
+        assert_eq!(agg_changeset, Some(changeset)); // TODO assert
```

Comment:
Is there something else this TODO comment is referring to?

Reply:
Good question, I think I put that there before I had the assert and forgot to remove it.